### DISCUSSION TITLE
bulge_to_arc how does the start/end works

### DISCUSSION THREAD

**Discussion by LoaBe** [mozman/ezdxf#409] [Q&A]
Hello,

I'm currently parsing a DXF file and when I reach a polyline I use the bulge_to_arc function to draw the curve part of the polyline. The problem is that I don't understand the needed parameter because the function needs a start and stop point. but the coordinate linked to the bulge IE:
(3963.873871383152, 1609.472358049532, 0.0, 0.0, -0.0075356760223999)

Does this coordinate the start point of this bulge? And the endpoint will be the XY coordinates of the next polyline coordinates?

Sorry for this question I'm a complete newbie with this lib.

Regards,
Loan

**Comment by mozman** [comment]
The arc starts at the vertex which includes the bulge value and ends at the following vertex.

Added this information also to the [docs](https://ezdxf.mozman.at/docs/dxfentities/lwpolyline.html#bulge-value).

FYI: 

You can access virtual LINE and ARC entities of LWPOLYLINE entities: https://ezdxf.mozman.at/docs/dxfentities/lwpolyline.html#ezdxf.entities.LWPolyline.virtual_entities

Virtual entity means, the ARC and LINE entities are not assigned to any layout.

```Python
arc_and_lines = list(pline.virtual_entities())
```
For approximation (flattening) you can convert the LWPOLYLINE into a Path object: https://ezdxf.mozman.at/docs/path.html

```Python
from ezdxf import path

p = path.make_path(pline)
vertices = p.flattening(0.01)
```

**Reply by LoaBe** [reply]
Thank you for your answer. Now I can handle the arc but I still have a problem regarding the anti-clockwise angle.  As the function will always return a counter-clockwise virtual point a break will be made on the line. Do you know any method to recoin an anti-clockwise angle from a clockwise one? See attachment for further detail (the problematic angle are marked with purple arrows).

Thank you again for your help 
![MicrosoftTeams-image](https://user-images.githubusercontent.com/67980218/116438854-2ca7a880-a84f-11eb-8052-71c89fafc588.png)

**Comment by mozman** [comment]
I am not sure what you want to know? The opposite orientation of start angle  to end angle is end angle to start angle.
